### DISCUSSION TITLE
I am trying to add a handler which handles upload of csv file via `multipart`, but it doesnt work, please help

### DISCUSSION THREAD

**Discussion by kirillsalykin** [juhaku/utoipa#1149] [Q&A] [answered]
I was following the example, and added this code:
```
#[derive(ToSchema)]
struct EmailUpload {
    #[schema(content_media_type = "application/octet-stream")]
    email: Vec<u8>,
}

#[utoipa::path(
        post,
        request_body(content = inline(EmailUpload), content_type = "multipart/form-data"),
        path = "",
        responses(
            (status = 200)
        ),
    )]
async fn upload_emails(
    State(app_state): State<AppState>,
    mut multipart: Multipart,
) -> impl IntoResponse {
    if let Some(mut field) = multipart.next_field().await.unwrap() {
        println!("UPLOAD: {:?}", field);
        let name = field.name().unwrap().to_string();
        // TODO: save it to tempfile when big, for now it is ok too keep it in memory
        let data = field.text().await.unwrap();
        println!("DATA: {:?}", data);
    }

    StatusCode::NO_CONTENT
}

```
But the swagger doesnt offer me to upload file, it just treats `emails`as Vec of ints.

I would expect to have similar openapi (from here https://swagger.io/docs/specification/v3_0/describing-request-body/multipart-requests/)
```
requestBody:
  content:
    multipart/form-data: # Media type
      schema: # Request payload
        type: object
        properties: 
          profileImage: 
            type: string
            format: binary
```
But I dont know how to express this, please help

Ideally, I'd like to have a Form struct which has file field in it (with content of upload file), but I am totally OK to handle `Multipart` object form `axum`.

Thanks in advance!

**Comment by juhaku** [comment, marked as answer]
If you want to define the schema according to OpenAPI 3.0, you can override the type with `value_type` and then define `format` which should correlate to the OpenAPI formats. 
```rust
#[derive(ToSchema)]
struct EmailUpload {
    #[schema(content_media_type = "application/octet-stream", value_type = String, format = Binary)]
    email: Vec<u8>,
}
```

**Reply by kirillsalykin** [reply]
Worked, thank you!

Is there any integration on this with `axum` or I should handle `Multipart` myself?

**Reply by juhaku** [reply]
So far this need to be handled manually, there is no specific integration with multipart libraries. There is an example in `axum_typed_multipart` library, but that is also handling the multipart manually and targeting 4.2.x version of utoipa. Integrations with multipart libraries is something that need to be assessed in future.

**Reply by kirillsalykin** [reply]
clear, thank you for your help and great library!

have a good one

**Reply by juhaku** [reply]
Thanks you too.
